src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQMessageHostBuilderExtensions.cs

```diff
@@ -24,9 +27,9 @@ public static class RabbitMQMessageHostBuilderExtensions
         public static void AddRabbitMQWithConfig<T>(this IPublisherBuilder<T> builder, IConfiguration config) where T : notnull
         {
             builder.AddTransient<ICloudEventFormatter, JsonEventFormatter>();
-            builder.AddTransient<IRabbitMQConnectionFactory, RabbitMQConnectionFactory>();
-            builder.AddPublisher<RabbitMQMessagePublisher<T>>();
             builder.Configure<RabbitMQPublisherOptions<T>>(config);
+            builder.AddSingleton(sp => RabbitMQConnectionFactory<T>.From(sp.GetRequiredService<IOptions<RabbitMQConsumerOptions<T>>>().Value));
```

Comment:
Are you sure that we need `RabbitMQ*Consumer*Options` here?

Reply:
yes, because the `RabbitMQConsumerOptions<T>` contains the name of the queue the monitor is observing. I initially didn't plan to make any connection  property accessible in the `IRabbitMQConnectionFactory<T>` and therefore required this information. Now that the factory already exposes some information (to match the original unit tests) I could also add the name of the queue as read-only property to the factory and get rid of the options if you prefer.

Reply:
No need to do that. It only caught my attention that in a code block where basically every term starts with `RabbitMQPublisher` there was one called `RabbitMQConsumer[...]`.

Reply:
I checked it and it's not even possible because the `IConnectionFactory` is of course not aware of the actual queue because the queue is declared with a channel after the connection is established